src/App.tsx

```diff
@@ -1,61 +1,112 @@
+import React, { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setuserId] = useState('');
+  const [updatedTodos, setUpdatedTodos] = useState(todos);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+
+  const handleSubmit: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    if (title && userId) {
+      const newTodo = {
+        id: Math.max(...updatedTodos.map(todo => todo.id)) + 1,
+        title,
+        completed: false,
+        userId: +userId,
+        user: getUser(+userId),
+      };
+
+      setUpdatedTodos([...updatedTodos, newTodo]);
+      setFormSubmitted(false);
+      setTitle('');
+      setuserId('');
+    } else {
+      setFormSubmitted(true);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" onSubmit={handleSubmit} method="POST">
+
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            Title:
+            {' '}
+            <input
+              id="title"
+              name="title"
+              type="text"
+              placeholder="Enter a title"
+              value={title}
+              onChange={(event) => setTitle(event.target.value)}
+              data-cy="titleInput"
+            />
+          </label>
+
+          {(!title && formSubmitted) && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label htmlFor="title">
+            User:
```

Comment:
The label attribute 'htmlFor' should be set to 'user', not 'title'. Change htmlFor='title' to htmlFor='user'.